File: dataset/output-bk/MISTRAL_codestral-latest/Task179_MISTRAL_codestral-latest.java
import java.util.*;

public class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        int n = nums.length;
        int[] result = new int[n - k + 1];
        Deque<Integer> window = new LinkedList<>();

        for (int i = 0; i < n; i++) {
            while (!window.isEmpty() && window.peek() <= i - k) {
                window.poll();
            }
            while (!window.isEmpty() && nums[window.peekLast()] < nums[i]) {
                window.pollLast();
            }
            window.offer(i);
            if (i >= k - 1) {
                result[i - k + 1] = nums[window.peek()];
            }
        }

        return result;
    }
}

File: dataset/output/MISTRAL_codestral-latest/Task179_MISTRAL_codestral-latest.cpp
#include <vector>
#include <deque>

std::vector<int> maxSlidingWindow(std::vector<int>& nums, int k) {
    int n = nums.size();
    std::vector<int> result(n - k + 1);
    std::deque<int> window;

    for (int i = 0; i < n; i++) {
        while (!window.empty() && window.front() <= i - k) {
            window.pop_front();
        }
        while (!window.empty() && nums[window.back()] < nums[i]) {
            window.pop_back();
        }
        window.push_back(i);
        if (i >= k - 1) {
            result[i - k + 1] = nums[window.front()];
        }
    }

    return result;
}